### PR TITLE
[Doc] Add kerberos support in CREATE ROUTINE LOAD

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Why I'm doing:

What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 1]
```diff
@@ -210,6 +213,30 @@ If `property.group.id` is not specified, StarRocks generates a random value base
     "property.sasl.password" = "xxxxxx"
     ```
 
+  - Access Kafka using the SASL_PLAINTEXT security protocol and SASL/GSSAPI (Kerberos) authentication mechanism:
+
+    ```sql
+    "property.security.protocol" = "SASL_PLAINTEXT", -- Specifies the security protocol as SASL_PLAINTEXT
+    "property.sasl.mechanism" = "GSSAPI", -- Specifies the SASL authentication mechanism as GSSAPI. Default value is GSSAPI.
+    "property.sasl.kerberos.service.name" = "kafka", -- Specifies the broker service name, default is Kafka.
```
Comment (amber-create) [inline docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:221]: image
-> Default value is xxx.

File: docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 2]
```diff
@@ -188,26 +188,53 @@ FROM <data_source>
...
+    "property.ssl.certificate.location" = "FILE:client.pem", -- Client 的 public key 的位置
+    "property.ssl.key.location" = "FILE:client.key", -- Client 的 private key 的位置
+    "property.ssl.key.password" = "abcdefg" -- Client 的 private key 的密码
+    ```
+
+- 访问 Kafka 时，使用 SASL_PLAINTEXT 安全协议和 SASL/PLAIN 认证机制
+
+    ```sql
+    "property.security.protocol" = "SASL_PLAINTEXT", -- 指定安全协议为 SASL_PLAINTEXT
+    "property.sasl.mechanism" = "PLAIN", -- 指定 SASL 认证机制为 PLAIN
```
Comment (amber-create) [inline docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:210]: image
The English says "GSSAPI", which differs from the Chinese which says "PLAIN"

**Review by amber-create — DISMISSED** [review]

File: docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 3]
```diff
@@ -188,26 +188,53 @@ FROM <data_source>
...
+    "property.sasl.password" = "admin" -- SASL 的密码
+    ```
+
+- 访问 Kafka 时，使用 SASL_PLAINTEXT 安全协议和 SASL/GSSAPI (Kerberos) 认证机制
+
+  ```sql
+  "property.security.protocol" = "SASL_PLAINTEXT", -- 指定安全协议为 SASL_PLAINTEXT
+  "property.sasl.mechanism" = "GSSAPI", -- 指定 SASL 认证机制为 GSSAPI, 默认是 GSSAPI
+  "property.sasl.kerberos.service.name" = "kafka", -- 指定 broker service name，默认是 Kafka
+  "property.sasl.kerberos.keytab" = "/home/starrocks/starrocks.keytab", -- 指定 client keytab location
```
Comment (evelynzhaojie) [inline docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:221]: “指定 client keytab 的位置”，和上面示例一样，用中文

File: docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 4]
```diff
@@ -181,6 +181,9 @@ If `property.group.id` is not specified, StarRocks generates a random value base
   - PLAIN
   - SCRAM-SHA-256 and SCRAM-SHA-512
   - OAUTHBEARER
+  - GSSAPI (Kerberos)
+
+  **Examples:**
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:186]: a line of code is missing after line 193

File: docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 5]
```diff
@@ -210,6 +213,30 @@ If `property.group.id` is not specified, StarRocks generates a random value base
     "property.sasl.password" = "xxxxxx"
     ```
 
+  - Access Kafka using the SASL_PLAINTEXT security protocol and SASL/GSSAPI (Kerberos) authentication mechanism:
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:216]: align the code comments with original code examples
image

File: docs/en/loading/RoutineLoad.md [hunk 6]
```diff
@@ -66,7 +66,7 @@ Routine Load now supports consuming CSV, JSON, and Avro (supported since v3.0.1)
 
          > **NOTE**
          >
-         > StarRocks supports access to Kafka via a security authentication mechanism SASL_SSL, SASL or SSL, or without authentication. This topic takes connection to Kafka without authentication as an example. If you need to connect to Kafka via a security authentication mechanism, see [CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md).
+         > StarRocks supports access to Kafka via a security protocol SASL_SSL, SAS_PLAINTEXT, SSL, or PLAINTEXT. This topic uses PLAINTEXT to connect to Kafka as an example. If you need to connect to Kafka via other security protocol, see [CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md).
```
Comment (evelynzhaojie) [inline docs/en/loading/RoutineLoad.md:69]: StarRocks supports access to Kafka via security protocols including SASL_SSL, SAS_PLAINTEXT, SSL, and PLAINTEXT. This topic uses connecting to Kafka via PLAINTEXT as an example. If you need to connect to Kafka via other security protocols, xxx

File: docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 7]
```diff
@@ -181,35 +181,54 @@ If `property.group.id` is not specified, StarRocks generates a random value base
...
+
+  **Examples:**
 
   - Access Kafka using the SSL security protocol:
 
     ```SQL
-    -- Specify the security protocol as SSL.
-    "property.security.protocol" = "ssl"
-    -- File or directory path to CA certificate(s) for verifying the kafka broker's key.
+    "property.security.protocol" = "ssl", -- Specifies the security protocol as SSL.
```
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:191]: Specify

File: docs/zh/loading/RoutineLoad.md [hunk 8]
```diff
@@ -68,7 +68,7 @@ Routine Load 目前支持从 Kakfa 集群中消费 CSV、JSON、Avro (自 v3.0.1
 
       > **说明**
       >
-      > StarRocks 可以通过安全认证机制，包括 SASL_SSL 认证、SASL 认证和 SSL 认证，以及无认证的方式连接 Kafka。本文以无认证的方式连接 Kafka 为例进行说明，如果您需要通过安全认证机制连接 Kafka，请参见[CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md)。
+      > StarRocks 可以通过安全协议，包括 SASL_SSL、SAS_PLAINTEXT、SSL，或者 PLAINTEXT。本文以 PLAINTEXT 方式连接 Kafka 为例进行演示，如果您需要通过其他安全协议连接 Kafka，请参见[CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md)。
```
Comment (evelynzhaojie) [inline docs/zh/loading/RoutineLoad.md:71]: 第一句不太完整：StarRocks 支持通过安全协议，包括 SASL_SSL、SAS_PLAINTEXT、SSL，或者 PLAINTEXT 来连接 Kafka。

File: docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md (hunk 7 above)
Comment (evelynzhaojie) [inline docs/en/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:191]: Specify -- 命令注释里用动词原形，下面相同。

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport  branch-3.2 branch-3.1 branch-3.0 branch-2.5

**Pull request merged**